plat/kvm/Config.uk

```diff
@@ -126,35 +118,66 @@ config KVM_DEBUG_VGA_CONSOLE
         help
           Choose VGA console for the debug printing
 
+config KVM_KERNEL_SERIAL_CONSOLE
+        bool "Serial console for the kernel prints"
+        default y
+        depends on (ARCH_X86_64 || ARCH_ARM_64)
+        help
+          Choose serial console for the kernel printing
+
 if (KVM_KVM_KERNEL_SERIAL_CONSOLE || KVM_DEBUG_SERIAL_CONSOLE)
+
+choice
+	prompt "Serial console driver"
+	depends on ARCH_ARM_64
+	default UART_PL011 if KVM_VMM_QEMU
+	default UART_NS16550 if KVM_VMM_FIRECRACKER
+
+config UART_PL011
+	bool "PrimeCell UART (PL011)"
+
+config UART_NS16550
+	bool "8250 / NS16550"
+endchoice
+
 menu "Serial console configuration"
-	if ARCH_X86_64
-	choice
-		prompt "Baudrate"
-		default KVM_SERIAL_BAUD_115200
-
-		config KVM_SERIAL_BAUD_115200
-			bool "115200"
-
-		config KVM_SERIAL_BAUD_57600
-			bool "57600"
-
-		config KVM_SERIAL_BAUD_38400
-			bool "38400"
-
-		config KVM_SERIAL_BAUD_19200
-			bool "19200"
-	endchoice
-	endif
-
-	if ARCH_ARM_64
-	config EARLY_PRINT_PL011_UART_ADDR
-		hex "Early debug console pl011 serial address"
-		default 0x09000000
-		depends on (KVM_DEBUG_SERIAL_CONSOLE && ARCH_ARM_64)
-		help
-			Pl011 serial address used by early debug console.
-	endif
+if ARCH_X86_64
+choice
+	prompt "Baudrate"
+	default KVM_SERIAL_BAUD_115200
+
+	config KVM_SERIAL_BAUD_115200
+		bool "115200"
+
+	config KVM_SERIAL_BAUD_57600
+		bool "57600"
+
+	config KVM_SERIAL_BAUD_38400
+		bool "38400"
+
+	config KVM_SERIAL_BAUD_19200
+		bool "19200"
+endchoice
+endif
+
+config EARLY_PRINT_PL011_UART_ADDR
+	hex "Early debug console pl011 serial address"
+	depends on UART_PL011
+	default 0x09000000
+	depends on (KVM_DEBUG_SERIAL_CONSOLE && ARCH_ARM_64)
+	help
+		Pl011 serial address used by early debug console.
+
+config EARLY_UART_NS16550
+	bool "Early console"
+	depends on UART_NS16550
+	default n
+
+config EARLY_UART_NS16550_BASE
+	hex "Early debug console ns16550 serial address"
+	depends on EARLY_UART_NS16550
+	help
```

Comment:
For Firecracker it is not possible as it lays out the MMIO regions depending on the devices to use. Saying that, there should be no default value for any driver. That is because a given driver can be potentially used by different platforms. In other words, the address for early UART only makes sense as a command-line parameter.
The only reason I am leaving the default for `pl011` is backwards compatibility. In the scope of #986 we will be removing this too.